### DISCUSSION TITLE
Serverless driver for javascript performance

### DISCUSSION THREAD

**Discussion by rcoundon** [planetscale/discussion#311] [Q&A]
Has anyone used the [PlanetScale serverless driver for Javascript](https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript)? I just tried out it out (using [kysely-planetscale](https://github.com/depot/kysely-planetscale)) and did a small performance comparison with the standard MySQL connection and the new driver in a lambda function.

I found that when using the serverless driver the cold start times and the execution times in a warm lambda take at least twice as long, often longer.

I haven't yet added any logging of what's taking the additional time, but before I dig into things further I was curious to find out what others' experiences were in terms of performance?

**Comment by rcoundon** [comment]
To help illustrate what's happening, here's the trace with the serverless driver
![image](https://user-images.githubusercontent.com/9136570/196748344-3b392475-ab3d-4fa1-91fd-0999474e9bc9.png)

and here with mysql driver
<img width="1534" alt="image" src="https://user-images.githubusercontent.com/9136570/196760571-787f478c-733c-48c5-841d-6b41093b6983.png">


As you can see, after the query returns in the mysql version, the next API calls to another service happen immediately, whereas with the serverless driver, there's a big time gap. (The post to psdb.v1alpha1... is the PlanetScale query - I'm sure that's obvious to you but just wanted to be specific!)

This smells like an event loop problem in our own code (we're using Node.js) but the only code that's changed is to use the driver and provide got-fetch as the HTTP client rather than mysql native.

**Reply by orware** [reply]
Hi @rcoundon,

I don't have any direct insight that might be happening but it's certainly an interesting observation! 

I wonder why there is such a gap between that `POST` request and the subsequent `GET` request that follows it in your first screenshot above. 

If you learn anything that explains it I'm sure it will be helpful for others so we appreciate the additional investigation you may be working through there on your end to figure out the "why", but please keep us updated either way!

**Comment by mattrobenolt** [comment]
And just to mirror the conversation over here since this is happening in multiple places: https://github.com/planetscale/database-js/issues/69

We have some back and forth on the actual issue. cc @orware

**Comment by rcoundon** [comment]
Yeah, apologies for posting in multiple places, I wasn't sure what the preference was - a discussion or an issue. Happy to close both, I can still back and update if I get to the bottom of things
